org.eclipse.winery.yaml.common/src/main/java/org/eclipse/winery/yaml/common/validator/TypeValidator.java

```diff
@@ -99,17 +100,17 @@ public Result visit(TCapabilityType node, Parameter parameter) {
 
     @Override
     public Result visit(TDataType node, Parameter parameter) {
-        if (node.getDerivedFrom() != null && !validateTypeIsDefined(node.getDerivedFrom(), typeVisitor.getDataTypes())) {
```

Comment:
Wouldn't it be an improvement, if you also extract all the code clones into a private method like illustrated in the following?
```
private Result visitElement(TInterfaceType node, P parameter) {
    if (node.getDerivedFrom() != null && !validateTypeIsDefined(node.getDerivedFrom(), typeVisitor.getInterfaceTypes())) {
        String msg = "The parent type \"" + node.getDerivedFrom() + "\" is undefined! \n" + print(parameter.getContext());
        setException(new InvalidParentType(msg));
    }
    super.visit(node, parameter);
    return null;
}
```

Reply:
Validation of parent type for all TOSCA Elements extending TEntityType
```
private void validateEntityType(TEntityType node, Parameter parameter, Map<String, List<String>> types) {...}
```
should solve this request